libs/ardour/session_state.cc

```diff
@@ -650,6 +650,8 @@ Session::create (const string& session_template, BusProfile const * bus_profile,
 		}
 
 	} else {
+		set_superclock_ticks_per_second (56448000); /* 2^10 * 3^2 * 5^3 * 7^2 */
```

Comment:
Old sessions (v6 or earlier) may not have a super-clock as part of the session-state.
Superclock needs to be set in that case

Reply:
A very high level comment:
I haven't been able to spot a single point where I can say that HERE all global(ish) variables have been set. There is not even a single file that takes care of all cases. New sessions and existing sessions are handled in very different ways. Admittedly, it works, and all reported bugs have been fixed. (Even though I think some bugs still are lurking if creating "new" from an existing session, and some global values are inherited.)
These changes are trying to clean it up a bit, but without proposing architectural changes that I wouldn't dare to propose.

This function is for creating *new* sessions. And this chunk is in the code path is for *not* using a template. This is the point where the global tempomap is initialized, and it seems like it should be an invariant that right before doing that, the superclock should be (re)initialized. The comment about old sessions thus doesn't seem trivially applicable. Please elaborate if it still applies. In either case, it seems like supportive evidence that these global variables (and their initialization if we can't get rid of them) could benefit from a cleanup.

Reply:
> This function is for creating new sessions.

You can create new sessions from templates

Reply:
Right. This PR has a problem there ... but the problem is even more general.

New sessions based on templates are handled by copying the template to the session file, then calling Session::load_state on it as if it was any other existing session. 

But I realize that this PR fails to handle setting superclock if loading an "old" v6 session that doesn't have superclocks-per-second. 46117303da7 not only fixed potentially uninitialized variables; it also fixed a real problem of loading sessions without superclock, falling back to the existing global default.

That would need fixing.